swifttool/manager.py

```diff
@@ -0,0 +1,288 @@
+                        updated_cfg['zones'][zone][ip]['disks'][builder] = \
+                            newdevs
+            if changed:
+                ringsdef_helper(updated_cfg, self.metadata, self.outdir)
+            return changed, updated_cfg
+        except Exception as e:
+            print "Error updating cluster configuration", e
+            sys.exit(-1)
+
+    @staticmethod
+    def _get_zone_ip(devs):
+        for dev in devs:
+            if 'ip' in dev:
+                return dev['ip']
+
+    def _calculate_cluster_changes(self):
+        try:
+            self.cluster_config = yaml.load(open(self.def_file, 'r'))
+            expected_config = get_devices(self.cluster_config['zones'])
+            current_config = self._get_all_cluster_devices()
+            # To get devices to be removed we look for devices present in
+            # current_config and not in expected_config
+            # To get list of devices to be added look for devices in expected
+            # not in current
+            if self.scaleup:
+                changes = self._get_devices_changed(expected_config,
+                                                    current_config)
+            else:
+                changes = self._get_devices_changed(current_config,
+                                                    expected_config)
+            self._normalize_changes(changes)
+            # Each ring will be list of devices to be modified
+            # Each device (in ring) will have following attributes:
+            # - zone
+            # - ip
+            # - device
+            # - target (device weight when adding, 0 when removing)
+            # - step (amount to change every time till target reached)
+            # step is defaulted to 10% or -10% of target
+        except Exception as e:
+            print "Error getting device info", e
+            sys.exit(-1)
+
+    def _normalize_changes(self, source):
+        for builder in RING_TYPES:
+            for zone, devadd in source[builder].items():
+                self.cluster_changes.setdefault(builder, [])
+                devices = []
+                for device in devadd:
+                    dev = {}
+                    dev['ip'] = device['ip']
+                    dev['device'] = device['device']
+                    if 'metadata' in device:
+                        dev['metadata'] = device['metadata']
+                    if 'delta' in device:
+                        step = int(math.ceil(device['delta'] /
+                                             self.iterations))
+                    else:
+                        step = int(math.ceil(device['weight'] /
+                                             self.iterations))
+                    if self.scaleup:
+                        dev['target'] = device['weight']
+                        dev['step'] = step
+                    else:
+                        dev['target'] = 0
+                        dev['step'] = int(-1 * step)
```

Comment:
Why cast this to an `int` again?  Aren't we guaranteeing this is already going to evaluate to an `int` because of the explicit casts above?